### PR TITLE
[5.0] Create 'make:seed' command to speed up the workflow

### PR DESCRIPTION
**Pull request by gil0mendes**
This command creats a file on 'database/seeds' with the name
'{{name}}TableSeeder'. After file creation composer dump-autoload
command is executed.

### CODE REVIEW

File: src/Illuminate/Database/Console/SeedMakeCommand.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+			array('name', InputArgument::REQUIRED, 'The name of the seeder'),
+		);
+	}
+
+	/**
+	 * Get the console command options.
+	 *
+	 * @return array
+	 */
+	protected function getOptions()
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/SeedMakeCommand.php]: Remove this method.

File: src/Illuminate/Database/Console/SeedMakeCommand.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
+<?php namespace Illuminate\Database\Console;
+
+use Illuminate\Console\Command;
+use Illuminate\Foundation\Composer;
+use Illuminate\Database\Seeder\SeedCreator;
+use Symfony\Component\Console\Input\InputOption;
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/SeedMakeCommand.php]: Remove all unused imports.

File: src/Illuminate/Database/Seeder/SeedCreator.php [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+     * The registered post create hooks.
+     *
+     * @var array
+     */
+    protected $postCreate = array();
+
+    /**
+     * Create a new seed creator instance
+     *
+     * @param \Illuminate\Filesystem\Filesystem $files
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/SeedCreator.php]: Incorrect spacing.

File: src/Illuminate/Database/Seeder/SeedCreator.php [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+	 * Get the seeder stub file.
+	 *
+	 * @return string
+	 */
+	protected function getStub()
+	{
+		return $this->files->get($this->getStubPath().'/blank.stub');
+	}
+
+    /**
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/SeedCreator.php]: ?

File: src/Illuminate/Database/Seeder/SeedCreator.php [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	 *
+	 * @param  string  $name
+	 * @return string
+	 */
+	protected function getClassName($name)
+	{
+		return studly_case($name);
+	}
+
+    /**
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/SeedCreator.php]: ?

File: src/Illuminate/Database/Seeder/SeedCreator.php [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+	 *
+	 * @return void
+	 */
+    protected function firePostCreateHooks()
+    {
+        foreach ($this->postCreate as $callback)
+		{
+			call_user_func($callback);
+		}
+    }
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/SeedCreator.php]: ?

File: src/Illuminate/Database/Seeder/SeedCreator.php [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+    /**
+     * Register a post seeder create hoook.
+     *
+     * @param \Closure $callback
+     * @return void
+     */
+    public function afterCreate(Closure $callback)
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/SeedCreator.php]: This whole method's not indented correctly.

File: src/Illuminate/Database/Seeder/SeedCreator.php [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    /**
+     * Get the full path to the seeder.
+     *
+     * @param  string  $name
+     * @param  string  $path
+     * @return string
+     */
+    public function getPath($name, $path)
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/SeedCreator.php]: This whole method's not indented correctly.

File: src/Illuminate/Database/Seeder/SeedCreator.php [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+    {
+        return $path.'/'.$name.'TableSeeder.php';
+    }
+
+    /**
+     * Get the path to the stubs.
+     *
+     * @return string
+     */
+    public function getStubPath()
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/SeedCreator.php]: This whole method's not indented correctly.

File: src/Illuminate/Database/Seeder/blank.stub [hunk 10]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+use Illuminate\Database\Seeder;
+
+class {{class}} extends Seeder {
+
+    /**
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/blank.stub]: This entire class needs to be indented with tabs.

File: src/Illuminate/Database/Console/SeedMakeCommand.php (hunk 1 above)
Comment (gil0mendes) [inline src/Illuminate/Database/Console/SeedMakeCommand.php]: Only the getOptions() or the getArguments() too?

Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/SeedMakeCommand.php]: No, why would you remove the other one?

Comment (gil0mendes) [inline src/Illuminate/Database/Console/SeedMakeCommand.php]: Another question, the declaration of an array can be `[]` or Laravel continues to prefer `array()`???

Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/SeedMakeCommand.php]: It doesn't matter what you use. :)

**Comment by gil0mendes** [comment]
@GrahamCampbell I made the changes you have asked

File: src/Illuminate/Database/Seeder/SeedCreator.php (hunk 3 above)
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/SeedCreator.php]: Still incorrect.

**Comment by GrahamCampbell** [comment]
Fix the remaining issue then squash to one commit please.

**Comment by gil0mendes** [comment]
I never did a squash before. Probably did wrong :/

**Comment by GrahamCampbell** [comment]
Looks like you forgot to force push. Basically, if there's not exactly 1 commit here, then you did it wrong.

**Comment by gil0mendes** [comment]
Check now, please

File: src/Illuminate/Database/Seeder/SeedCreator.php (hunk 3 above)
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/SeedCreator.php]: Still incorrect.

File: src/Illuminate/Database/Seeder/SeedCreator.php [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+     */
+    public function __construct(Filesystem $files)
+    {
+        $this->files = $files;
+    }
+
+    /**
+     * Create a new seeder at the given path.
+     *
+     * @param  string $name
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/SeedCreator.php]: incorrect

File: src/Illuminate/Database/Seeder/SeedCreator.php [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+     */
+    protected function getStub()
+    {
+        return $this->files->get(__DIR__ . '/blank.stub');
+    }
+
+    /**
+     * Populate the place-holders in the seeder stub.
+     *
+     * @param  string $name
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/SeedCreator.php]: incorrect

File: src/Illuminate/Database/Seeder/SeedCreator.php [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        $stub = str_replace('{{class}}', studly_case($name) . 'TableSeeder', $stub);
+
+        return $stub;
+    }
+
+    /**
+     * Get the full path to the seeder.
+     *
+     * @param  string $name
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Seeder/SeedCreator.php]: incorrect

File: src/Illuminate/Database/Console/SeedMakeCommand.php [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+		$this->composer->dumpAutoloads();
+	}
+
+	/**
+	 * Write the seeder file to disk.
+	 *
+	 * @param  string $name
+	 * @param  string $table
+	 * @param  bool $create
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/SeedMakeCommand.php]: Please look at laravel's docblocks because you've got them wrong over and over.

Comment (gil0mendes) [inline src/Illuminate/Database/Console/SeedMakeCommand.php]: What still wrong in this block of code? I replace any spaces with tabs :/

Comment (RomainLanz) [inline src/Illuminate/Database/Console/SeedMakeCommand.php]: You need to add 2 spaces after the var type

**Comment by taylorotwell** [comment]
Overcomplicated, could have extended the GeneratorCommand the other generators use.

**Comment by gil0mendes** [comment]
I used the MigrateMakeCommand like my base.

**Comment by taylorotwell** [comment]
I would just copy and paste another seeder.

**Comment by JosephSilber** [comment]
@taylorotwell that can be said of the other generators too?

**Pull request closed without merging**